ui/src/app/machines/views/MachineDetails/MachineCommissioning/MachineCommissioning.tsx

```diff
@@ -36,10 +41,23 @@ const MachineCommissioning = (): JSX.Element => {
   );
 
   useEffect(() => {
-    if (!scriptResults?.length && !loading) {
+    if (
+      !loading &&
+      (!scriptResults?.length ||
+        // Refetch the script results when the commissioning status changes, otherwise
+        // the new script results won't be associated with the machine.
```

Comment:
This took me a while to wrap my head around. I keep assuming that a scriptresult's id is constant, but each run of a test script is an entirely new object. No issue with the comment - just my reading comprehension!

Also just wondering - do we want to refetch the script results on *all* status changes? Or is it only necessary when it changes to `RUNNING` (in which case there are new scriptresults)? Sorry, I'm pretty hazy with how scriptresults work.

Reply:
Yes you're right, except the new results get created when the status changes to `PENDING`. I just verified and that worked.